### DISCUSSION TITLE
Seq C# API best way to paginate logs.

### DISCUSSION THREAD

**Discussion by sirmeepington** [datalust/seq-tickets#2158] [Help] [answered]
Which is the best way to properly paginate log entries via the C# API? 

I've been using `EnumerateAsync` to get a list of logs of a certain count, but I can't figure out how to properly paginate the result in a way that doesn't involve gathering _every_ log entry and paginating on those.

**Comment by nblumhardt** [comment, marked as answer]
Hi @sirmeepington,

Both `EnumerateAsync()` accepts a `startAfterId` parameter that specifies the end of the previous page; once you've received a page of `count` entities, grab the `Id` of the last one, and pass this through when requesting the next page.

Let me know if you need any more info.

**Reply by sirmeepington** [reply]
Sounds like a good plan. I'll give that a go and I'll re-comment here if anything goes wrong :)
